install-dev/data/xml/hook.xml

```diff
@@ -11,6 +11,11 @@
       <title>New orders</title>
       <description/>
     </hook>
+    <hook id="actionValidateOrderAfter">
+      <name>actionValidateOrderAfter</name>
+      <title>New orders</title>
+      <description>This hook is called after the complete creation of an order</description>
```

Comment:
You're also able to call validateOrder outside the scope of creation of a new order

Reply:
Si un problème lors de validateOrder la commande n’est pas complète et ca peut provoquer des erreurs ou des soucis surtout lors de liaison avec des outils externes.
Ses liens sont déclenchés lors du passage d’une commande.
J’ai eu le cas ou une liaison changeait le statut pour mettre « envoyer à … pour traitement », sauf que le statut de la commande « paiement accepté » était ajouter après !
J’avais pour la commande « Envoyer à … pour traitement » et le statut en cours était « paiement accepté » alors que ca doit petre l’inverse.
Ce hook est exécuter à la toute fin du process d’enregistrement.

Reply:
It is after a long experience, very often the problem is that there is an error in the ValidateOrder and the order is not complete. Connections with ERP systems are sensitive. This hook allowed me to make the exchanges more reliable.

Reply:
I guess not, validateOrder is only called at order creation time by a payment module.

Reply:
In the prestashop context, but the hook itself has another context. Reading the method and what you can do with it, it should be 'After validating an order' or something similar